### PR TITLE
Fixes #350 Personalized homepage

### LINKED ISSUE #350: UR Finding: Users wanted personalized homepage
UR Finding: Users wanted personalized homepage - which included seeing their own Favorites.

### PR DESCRIPTION
**Pull request by keianhzo**
Fixes #350 Added a setting to set a homepage in the Developer options panel. Also we now provide an adb parameter to do it from the command line.

### CODE REVIEW

File: app/src/common/shared/org/mozilla/vrbrowser/browser/SessionStore.java [hunk 1]
```diff
@@ -492,15 +492,17 @@ public void setRegion(String aRegion) {
     }
 
     public String getHomeUri() {
-        String result = SessionStore.HOME_WITHOUT_REGION_ORIGIN;
+        String result = SettingsStore.getInstance(mContext).getHomepage();
         if (mRegion != null) {
-            result = SessionStore.HOME_WITHOUT_REGION_ORIGIN + "?region=" + mRegion;
+            result = result + "?region=" + mRegion;
```
[the remark this answers is not in the document]
Comment (keianhzo) [reply app/src/common/shared/org/mozilla/vrbrowser/browser/SessionStore.java]: Fixed, skipping if the URL is not the default homepage.

**Comment by keianhzo** [comment]
@cvan Thanks for the review! I have fixed the mentioned issues.

**Comment by cvan** [comment]
1. Autofocus text field for `Homepage` field when `Edit` button is pressed.

### Current
image
image

### Expected
(Also, see item 4 below re: `Firefox Home (Default)`)
image

<hr>

2. URL disappears when you **double-press** 2-3 times to the right of the text in the field. (The text and highlight colour appear both white.)

image

<hr>

3. Press+drag-and-holding on a segment of the text in the field should be **white on navy** instead of **dark gray on blue**.

image

Compare to double-pressing which shows **white on dark gray**:

image

<hr>

4. When `Homepage` is empty (`hint`) or the text value (default) is displayed, is it possible to print `Firefox Home (Default)` instead of `https://webxr.today`?

image

image

## Context
I propose the simplest option today would be, instead of exposing `https://webxr.today`, we use the same string that is used on Firefox desktop: `Firefox Home (Default)`

When we have multiple tabs, windows, and other features for user customisation, we can revisit how to augment this.

### Desktop
Firefox 64.0 macOS
image

### Mobile
Firefox 14.0 (12646) for iOS
img_6534
img_6535

**Comment by keianhzo** [comment]
@cvan Thanks one more time for the thorough review, have helped to fix some other pre existing bugs. All 1-5 issues above should be fixed and working as proposed.

**Comment by keianhzo** [comment]
@cvan 
1. I've set the hint for both the edit and non-edit modes, I think is more consistent that way.
2. I've fixed all the focus related changes, now everything should get closed when focus changes.

**Comment by cvan** [comment]
> 1. I've set the hint for both the edit and non-edit modes, I think is more consistent that way.

nice, the hint and text are both correct.

one more nit: is it possible to disable editing of the text when the value is `Firefox Home (Default)`? specifically, is there a way to disable character selection and force only select-all / clear?

looking to avoid this situation:

image
image
image

> 2. I've fixed all the focus related changes, now everything should get closed when focus changes.
this works perfectly! I love it.

**Pull request merged**